src/main/java/uk/gov/hmcts/ccd/domain/model/common/CaseReferenceUtils.java

```diff
@@ -11,4 +13,20 @@ public static String formatCaseReference(String caseReference) {
         }
         return caseReference;
     }
+
+    public static boolean checkRegex(final String value) {
+        final String regularExpression = "(?:^[0-9]{16}$|^\\d{4}-\\d{4}-\\d{4}-\\d{4}$)";
```

Comment:
can we have this as a constant as a class field please. 
Also, can you give it a better name that helps understand what the regex is ?